### PR TITLE
8473 add rules gcp vpc storage access

### LINKED ISSUE #8473: Add rules for GCP VPC Storage access logs
## Description

Add rules for Google Cloud Platform usage logs and storage logs. Logs operations and usage of the storage resources, well defined formats. Storage resource logs may be not of interest, further analysis needed.

Reference:
- https://cloud.google.com/storage/docs/access-logs

Usage log format:
| Field               | Type    | Description                                                                                                                                                                                                                                                                           |
| ------------------- | ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| time\_micros        | integer | [The time that the request was completed, in microseconds since the](https://en.wikipedia.org/wiki/Unix_epoch) [Unix epoch](https://en.wikipedia.org/wiki/Unix_epoch).                                                                                                                |
| c\_ip               | string  | The IP address from which the request was made. The "c" prefix indicates that this is information about the client.                                                                                                                                                                   |
| c\_ip\_type         | integer | The type of IP in the c\_ip field:A value of 1 indicates an IPV4 address.<br>A value of 2 indicates an IPV6 address.                                                                                                                                                                  |
| c\_ip\_region       | string  | Reserved for future use.                                                                                                                                                                                                                                                              |
| cs\_method          | string  | The HTTP method of this request. The "cs" prefix indicates that this information was sent from the client to the server.                                                                                                                                                              |
| cs\_uri             | string  | The URI of the request.                                                                                                                                                                                                                                                               |
| sc\_status          | integer | The HTTP status code the server sent in response. The "sc" prefix indicates that this information was sent from the server to the client.                                                                                                                                             |
| cs\_bytes           | integer | The number of bytes sent in the request.                                                                                                                                                                                                                                              |
| sc\_bytes           | integer | The number of bytes sent in the response.                                                                                                                                                                                                                                             |
| time\_taken\_micros | integer | The time it took to serve the request in microseconds, measured from when the first byte is received to when the response is sent. Note that for resumable uploads, the ending point is determined by the response to the final upload request that was part of the resumable upload. |
| cs\_host            | string  | The host in the original request.                                                                                                                                                                                                                                                     |
| cs\_referer         | string  | [The](https://en.wikipedia.org/wiki/HTTP_referrer) [HTTP referrer](https://en.wikipedia.org/wiki/HTTP_referrer) for the request.                                                                                                                                                      |
| cs\_user\_agent     | string  | The User-Agent of the request. The value is GCS Lifecycle Management for requests made by lifecycle management.                                                                                                                                                                       |
| s\_request\_id      | string  | The request identifier.                                                                                                                                                                                                                                                               |
| cs\_operation       | string  | The Cloud Storage operation e.g. GET\_Object.                                                                                                                                                                                                                                         |
| cs\_bucket          | string  | The bucket specified in the request. If this is a list buckets request, this can be null.                                                                                                                                                                                             |
| cs\_object          | string  | The object specified in this request. This can be null.                                                                                                                                                                                                                               |

Storage log format:
| Field                | Type    | Description                                                                                                                   |
| -------------------- | ------- | ----------------------------------------------------------------------------------------------------------------------------- |
| bucket               | string  | The name of the bucket.                                                                                                       |
| storage\_byte\_hours | integer | Average size in byte-hours over a 24 hour period of the bucket. To get the total size of the bucket, divide byte-hours by 24. |

Usage Logs:
```json
{
   "integration":"gcp",
   "gcp":{
      "time_micros":"1623657113850001",
      "c_ip":"34.75.50.245",
      "c_ip_type":"1",
      "c_ip_region":"",
      "cs_method":"GET",
      "cs_uri":"/download/storage/v1/b/framework-test-bucket/o/usage-logs_usage_2021_06_11_14_00_00_0deb315d529495e45e_v0?alt=media&generation=1623445458746803",
      "sc_status":"206",
      "cs_bytes":"0",
      "sc_bytes":"644",
      "time_taken_micros":"141000",
      "cs_host":"storage.googleapis.com",
      "cs_referer":"",
      "cs_user_agent":"DataflowBatchWorkerHarness Google-API-Java-Client/1.30.10 Google-HTTP-Java-Client/1.36.0 (gzip)",
      "s_request_id":"ABg5-UwwFhsTGuOn2WpJdUzroNMTNuh-IabN-9XKEwnEJMi3Hcv_o96mM8lqKU3wWWGko5RL4pEoaepuizcP_NQIKhw",
      "cs_operation":"storage.objects.get",
      "cs_bucket":"framework-test-bucket",
      "cs_object":"usage-logs_usage_2021_06_11_14_00_00_0deb315d529495e45e_v0",
      "source":"gcp_bucket"
   }
}
```

Storage Logs:
```json
--WIP--
```

## Tasks
- [ ] Add Usage rules
  - [ ] Generic use case rules
  - [ ] Create test file .ini
- [ ] Add storage rules
  - [ ] Generic use case rules
  - [ ] Create test file .ini

### PR DESCRIPTION
**Pull request by JavierBejMen**
|Related issue|
|---|
|Closes #8473 |

## Description
Added generic rules for GCP storage and usage logs.
Added tests in gcp.ini

## Tests

- Decoder/Rule tests
  - [x] Added unit testing files ".ini"
  - [x] runtests.py executed without errors

### CODE REVIEW

File: ruleset/rules/0690-gcp_rules.xml [hunk 1]
```diff
@@ -612,4 +612,18 @@ ID: 65000 - 65499
...
+        <description>GCP VPC Usage event</description>
+        <options>no_full_log</options>
+    </rule>
+
+    <!-- GCP VPC Storage rules-->
+    <rule id="65052" level="2">
+        <if_sid>65000</if_sid>
+        <field name="gcp.bucket">^(?!^$).+$</field>
+        <description>GCP VPC Storage event</description>
+
```
Comment (jcruzlp) [inline ruleset/rules/0690-gcp_rules.xml:628]: Missing `</rule>`

**Review by jcruzlp — CHANGES_REQUESTED** [review]

File: ruleset/rules/0690-gcp_rules.xml (hunk 1 above)
Comment (JavierBejMen) [reply ruleset/rules/0690-gcp_rules.xml:628]: :1st_place_medal:

**Review by jcruzlp — APPROVED** [review]
LGTM

**Review by 72nomada — APPROVED** [review]
LGTM

**Pull request merged**